### ISSUE TITLE
Proportion of infected plants vs proportion of infected quadrats?

### ISSUE THREAD

**Issue by IhsanKhaliq** [IhsanKhaliq/valascotraceR#5]
Is it worth trying to use proportion of infected plants instead of proportion of infected quadrats to see if we get better prediction? Both Art and Coventry used the former option. 
![Screen Shot 2021-11-11 at 1 49 43 pm](https://user-images.githubusercontent.com/54728693/141234091-c3be854f-14f3-44d3-9295-8c28ba5517a6.png)

![Screen Shot 2021-11-11 at 1 50 22 pm](https://user-images.githubusercontent.com/54728693/141234095-b80bab9a-1917-45ea-b932-043ea56c42c5.png)

![Screen Shot 2021-11-11 at 1 50 57 pm](https://user-images.githubusercontent.com/54728693/141234100-b74e6594-6249-4cd1-ad93-356b2b32ee44.png)
.

**Comment by adamhsparks** [comment]
That's more or less what's been done here. I took the proportion of infected growing points vs all growing points not proportions of quadrats.

**Comment by IhsanKhaliq** [comment]
Well we need proportion of infected quadrats or plants because we don't have proportion of infected growing points for observation. But I'm going through code and I'm not sure if we're following the same method to compare observation and prediction. For `prediction`, we're computing **proportion** of quadrats with infection  

https://github.com/IhsanKhaliq/valascotraceR/blob/f021bb884adc16a5f7f93cb07b45319e1e008c4d/example_trace_ascoRun.Rmd#L183-L188

For `observation`, we're computing the **mean** of `V1` column, and the column shows presence/absence of disease
https://github.com/IhsanKhaliq/valascotraceR/blob/f021bb884adc16a5f7f93cb07b45319e1e008c4d/example_trace_ascoRun.Rmd#L206-L210

Why are we computing `proportion` for predictions and `mean` for observation? We should be comparing proportion vs. proportion, not proportion vs. mean?
![Screen Shot 2021-11-15 at 3 43 22 pm](https://user-images.githubusercontent.com/54728693/141728749-0bc24531-3818-4f83-b780-d45fe47bfdae.png)

**Comment by adamhsparks** [comment]
I'll have to recheck, but if it worked as I intended, it should not be the proportion of infected quadrats out of those inspected but rather the proportion of infected growing points in the inspected quadrats.

**Comment by IhsanKhaliq** [comment]
You can do that for predictions but there is no way to compare that with the observation because we don't have infected growing points for observation. To compare with observation/disease curves, it should be either proportion of infected quadrats or infected plants

**Comment by adamhsparks** [comment]
OK, I misunderstood what you wanted in an earlier conversation. Previously, I was calculating the proportion of infected quadrats but you indicated that wasn't what was needed here, so I changed it to this.

Please make the appropriate changes as needed.

**Comment by IhsanKhaliq** [comment]
Ok we just need to replace `mean(v1)` to `sum(V1==1)/n())`

**Comment by adamhsparks** [comment]
Isn't that still the average?

**Comment by IhsanKhaliq** [comment]
> Isn't that still the average?

https://stackoverflow.com/questions/34792323/calculate-percentages-of-a-binary-variable-by-another-variable-in-r

**Comment by adamhsparks** [comment]
Ooops. I'm doing 50 things at once and misread. :) All good.

**Comment by adamhsparks** [comment]
I've replaced  `mean(v1)` with `sum(V1==1)/n())`. No major changes in the figures or fit that I can see.

**Comment by IhsanKhaliq** [comment]
I saw a very big difference earlier  (made it worst). I will need to re check my code tomorrow


________________________________

**Comment by adamhsparks** [comment]
Maybe it is worse. It's not better. They look more like what I'd initially had before I selected only the observed quadrats.

**Comment by adamhsparks** [comment]
OK, I think that this needs to be revised if you want to compare quadrats and not the level of infection in the quadrats.

```r
billa <-
  billa %>%
  filter(quadrat != "other") %>%
  mutate(infected = if_else(infectious_gp > 0, TRUE, FALSE, NA)) %>%
  group_by(i_date) %>%
  summarise(p_inf = sum(infected) / n())
```

This is comparing the number of growing points infected. Not the percent of quadrats that are infected.

**Comment by IhsanKhaliq** [comment]
I see what you mean. We need to create a new column 'infected` in which quadrats with `infectious_gp > 0` should be `1` and quadrats with 'infectious_gp <=0` should be zero.  Then we can do `sum(infected==1/n()` to calculate the proportion of infected quadrats?

**Comment by adamhsparks** [comment]
Yes, exactly, I poked at it a little bit last night but was too tired to make my brain work properly.

**Comment by IhsanKhaliq** [comment]
> Yes, exactly, I poked at it a little bit last night but was too tired to make my brain work properly.

Almost there Adam

**Comment by IhsanKhaliq** [comment]
I've made the change, but proportion of infected quadrats is the same. I think it's more or less the same thing, unless I'm doing something wrong. This vignette requires some code review because `V1` was not showing quadrats with `presence/presence` of disease as I thought. There might be some other discrepancies
![Screen Shot 2021-11-16 at 1 33 32 pm](https://user-images.githubusercontent.com/54728693/141915500-a908a126-8df0-45cb-8e7e-457d1c01ac2b.png)
![3](https://user-images.githubusercontent.com/54728693/141915649-3afe5cd7-b379-434a-9c73-248965ec6e5c.png)

**Comment by adamhsparks** [comment]
Yes, I've been struggling with exactly this too, @IhsanKhaliq. Glad I'm not alone. 

I'll see if I can create a clean script to import and summarise infected paddocks only and go from there and see if that helps me figure this out.

**Comment by adamhsparks** [comment]
Proportion of infected quadrats is now calculated and shown in the graph for modelled and observed data.

https://github.com/IhsanKhaliq/valascotraceR/commit/ab33d405f73fa9a6c1b2fff4c0acae375b131647

**Comment by IhsanKhaliq** [comment]
Awesome. Almost there. Not sure why heat map doesn't show disease spread though. . Data shows that 84 and 92 quadarats are infected in Billa and Tosari, respectively but heat map shows maybe 5% quadrat infection. Is it because we've taken mean instead of sum?

![Screen Shot 2021-11-16 at 8 18 49 pm](https://user-images.githubusercontent.com/54728693/141967266-d3d538da-ef8f-4e1d-a0e5-58b0202ee979.png)

![000003](https://user-images.githubusercontent.com/54728693/141967351-e3291fbc-5d93-4f4e-8386-f6671c48300d.png)

**Comment by IhsanKhaliq** [comment]
We may have to slightly tweak the spore_rate/primary inoculum intensity because the model slightly over predicted disease at mid season and slightly over predicted disease at the end of the season. But we are very close
![000003](https://user-images.githubusercontent.com/54728693/141967856-787ce407-e5d4-48f8-b13c-6c427b599ac1.png)

**Comment by adamhsparks** [comment]
Heat map shows mean per quadrat, so an average of 1 infected gp out of 20 runs in a quadrat is pretty small compared to the values at the focus. We can convert it to a binary infected/not or bin it into categories if you want so see the quadrat version of the line graph.


--
Dr Adam Sparks | Senior Research Scientist
Farming Systems Innovation
Primary Industries Development
Department of Primary Industries and Regional Development
3 Baron-Hay Court, South Perth WA 6151
t +61 (0)8 9368 3689 | m +61 (0)415 489 422 | w dpird.wa.gov.au

**Comment by IhsanKhaliq** [comment]
I think the line version will good because it is supposed to show spatial spread. The number of lesions per quadrat are already provided by `summarise_trace`?

**Comment by adamhsparks** [comment]
Using `lapply()` with `summarise_trace()` would give you total paddock daily values for the major growing point values that you’d then need to average for the 20 runs. It’s not been done here though.


--
Dr Adam Sparks | Senior Research Scientist
Farming Systems Innovation
Primary Industries Development
Department of Primary Industries and Regional Development
3 Baron-Hay Court, South Perth WA 6151
t +61 (0)8 9368 3689 | m +61 (0)415 489 422 | w dpird.wa.gov.au

**Comment by IhsanKhaliq** [comment]
For the resource announcement paper, we can get away with some sort of  table ( like hagis) if we want  

But for the model paper, we'd ideally need heatmap comparing obs vs. prediction.

Art heatmap
![Screen Shot 2021-11-16 at 8 30 30 pm](https://user-images.githubusercontent.com/54728693/141969460-05348a0c-59d6-4b44-941b-65b976e32139.png)

Coventry heatmap
![Screen Shot 2021-11-16 at 8 32 37 pm](https://user-images.githubusercontent.com/54728693/141969527-babf1f2f-97d7-4f29-80ae-e19a5ebe4b7a.png)

**Comment by IhsanKhaliq** [comment]
Or the line graph is also fine if we dont heatmap for the model paper. Can `summarise_trace` output be exported as a table for the package paper?

**Comment by adamhsparks** [comment]
Yes but you’d have 120 lines or however many rows there are for `i_day`


--
Dr Adam Sparks | Senior Research Scientist
Farming Systems Innovation
Primary Industries Development
Department of Primary Industries and Regional Development
3 Baron-Hay Court, South Perth WA 6151
t +61 (0)8 9368 3689 | m +61 (0)415 489 422 | w dpird.wa.gov.au

**Comment by IhsanKhaliq** [comment]
How you presented the output in your previously published paper, specifically resource announcement paper? Can we show first 10 days? Or something along those lines?

![images_medium_mpmi-07-19-0180-a_t1](https://user-images.githubusercontent.com/54728693/141971041-b0d9d694-d983-44cb-b8ad-74646af36490.gif)

**Comment by adamhsparks** [comment]
I guess you could. That table is the whole table in that announcement though.

I’d be showing the fit and agreement of the model and the observations though, the last figure. You want people to be convinced that it’s good enough to use it.


--
Dr Adam Sparks | Senior Research Scientist
Farming Systems Innovation
Primary Industries Development
Department of Primary Industries and Regional Development
3 Baron-Hay Court, South Perth WA 6151
t +61 (0)8 9368 3689 | m +61 (0)415 489 422 | w dpird.wa.gov.au

**Comment by IhsanKhaliq** [comment]
Let me search some relevant papers tomorrow then we can decide on this


________________________________

**Comment by IhsanKhaliq** [comment]
Closed ab33d405f73fa9a6c1b2fff4c0acae375b131647
